### ISSUE TITLE
Allowing to set endpoint differently for hasura console command and hasure console ui

### ISSUE THREAD

**Issue by Bessonov** [hasura/graphql-engine#5594]
I'm exploring hasura and faced in some missing setting. I use [vsc remote dev extension](https://code.visualstudio.com/docs/remote/remote-overview) extension with my own [dev-image](https://github.com/Bessonov/dev-image) and have following setup:

On remote host I run:
```
services:
  dev:
    ports:
      - "127.0.0.1:$BACKEND_PORT:3000" # nodejs backend
      - "127.0.0.1:$FRONTEND_PORT:8080" # react frontend
      - "127.0.0.1:$FRONTEND_PORT_2:8081" # hasura console port
      - "127.0.0.1:$FRONTEND_PORT_3:8097" # hasura console api port

  hasura:
    image: hasura/graphql-engine:v1.3.0
    ports:
      - "127.0.0.1:$HASURA_PORT:8085"
    environment:
      HASURA_GRAPHQL_DATABASE_URL: postgres://yugabyte:5433/hasura
      HASURA_GRAPHQL_ENABLE_CONSOLE: "false"
      [...]

  # other services like yugabyte db, file sync
```

From local host I connect with vsc into dev container. All commands are executed remotely in this container. Furthermore I pass remote ports to the local host with ssh forwarding. With this I'm able to run browser on local host and reach 3000, 8080, 8085 on local host.

Running console in hasura container works perfectly. But [with this step](https://hasura.io/docs/1.0/graphql/manual/migrations/migrations-setup.html#step-4-use-the-console-from-the-cli) I get some troubles. If I start the console in dev container:

```
# https://github.com/jjangga0214/hasura-cli version 1.3.0
pnpm run h -- console --address 0.0.0.0 --console-port 8081 --no-browser --log-level DEBUG --api-port 8097 
```
I get following result:

1. hasura console in dev container is able to reach the `endpoint: http://hasura:8085` through docker network.
2. I'm able to reach hasura console in local browser through http://0.0.0.0:8081/console/
3. But, the console ui try to communicate with http://hasura:8085/v1/query, which is not resolvable on local host.

As a workaround I've added `127.0.0.1 hasura` to `/etc/hosts` on local host and it starts working. It would be nice to have an option, to set endpoint for the ui to http://localhost:8085/ .

Side note: the option `--use-server-assets` still doesn't work for me and I get 404 on assets like http://hasura:8085/console/assets/common/css/font-awesome.min.css.gz .

**Comment by scriptonist** [comment]
Related: https://github.com/hasura/graphql-engine/issues/5594

**Comment by Bessonov** [comment]
@scriptonist that's an endless recursion :D

**Comment by ajbouh** [comment]
I'm experiencing exactly this issue as well. As far as I can tell:
- `--use-server-assets` assumes that `/console` is accessible on the hasura endpoint
- `--static-dir` just mounts the given directory at `/static` which seems... unrelated to anything that the console html ui actually uses?

**Comment by alexgorbatchev** [comment]
@rikinsk  I would also like that feature. I'm setting up a local k8s development cluster which exposes all services via `https://*.local.dev`, however `local.dev` points to `127.0.0.1`. I can get the console to bootup if i set endpoint to `http://graphql-engine:8080`, but then it tries to use the same URL from the browser, which isn't accessible because it's internal to kubernetes. If i set endpoint to `https://graphql-engine.local.dev` this doesn't work at all because `graphql-engine` isn't running via HTTPS.

Basically, being able to set `server-endpoint` and `browser-endpoint` separately would be very useful.
